tests/bears/BearTest.py

```diff
@@ -262,13 +269,44 @@ def test_new_result(self):
         expected = Result.from_values(bear, 'test message', '/tmp/testy')
         self.assertEqual(result, expected)
 
+
+class BearDownloadTest(BearTestBase):
+
+    def setUp(self):
+        super().setUp()
+        self.mock_url = 'https://test.com'
+        self.filename = 'test.html'
+        self.file_location = join(self.uut.data_dir, self.filename)
+
+    def test_connection_timeout_mocked(self):
+        exc = requests.exceptions.ConnectTimeout
+        with requests_mock.Mocker() as reqmock:
+            reqmock.get(self.mock_url, exc=exc)
+            with self.assertRaisesRegexp(exc, ''):
+                self.uut.download_cached_file(
+                    self.mock_url, self.filename)
+
+    def test_connection_failure(self):
+        exc = requests.exceptions.HTTPError
+        with self.assertRaisesRegexp(exc, '418 Client Error'):
+            self.uut.download_cached_file(
+                'http://httpbin.org/status/418', self.filename)
+
+    def test_read_timeout(self):
+        exc = requests.exceptions.ReadTimeout
+        with self.assertRaisesRegexp(exc, 'Read timed out.'):
+            self.uut.download_cached_file(
+                'http://httpbin.org/delay/10',
+                self.filename,
+                timeout=5)
```

Comment:
The test will consume 5 seconds.  I can shorten it a bit, and it should still work ok?

fwiw, it appears that mocking is not working ; see https://github.com/coala/coala/issues/3811#issuecomment-283214574